File: problems/003-binary-tree-serialization/java/NodeTests.java
import static org.junit.Assert.assertEquals;
import org.junit.Test;

public class NodeTests {
    
    @Test
    public void nodeTest() {
        // Arrange
        var node = new Node("root", new Node("left", new Node("left.left")), new Node("right"));
        // Act
        var result = Node.deserialize(Node.serialize(node)).left.left.val;
        // Assert
        assertEquals("left.left", result);
    }
}

File: problems/003-binary-tree-serialization/csharp/NodeTests.cs
using NUnit.Framework;

public class NodeTests
{
    [Test]
    public void NodeTest()
    {
        var node = new Node("root", new Node("left", new Node("left.left")), new Node("right"));

        var serialized = Node.Serialize(node);
        var deserialized = Node.Deserialize(serialized);

        var result = Node.Deserialize(Node.Serialize(node)).Left.Left.Val;

        Assert.That(result, Is.EqualTo("left.left"));
    }
}
